SConstruct

```diff
@@ -700,3 +700,16 @@ if "env" in locals():
     # TODO: replace this with `env.Dump(format="json")`
     # once we start requiring SCons 4.0 as min version.
     methods.dump(env)
+
+    import atexit
+
+    def check_build_failures():
+        failed = len(GetBuildFailures()) > 0
+        if failed and env["werror"] and not env.stable_release:
+            print(
+                "Build failed while treating warnings as errors. "
```

Comment:
How do you know from this check that the build failed because of `-Werror`?

Reply:
I haven't found a way to check whether a build has failed due to `-Werror`, but this message simply says that the build has failed, and while it failed, `werror` is detected, I didn't mean to imply causality in here.